qiskit/utils/backend_utils.py

```diff
@@ -156,7 +181,10 @@ def is_simulator_backend(backend):
     Returns:
         bool: True is a simulator
     """
-    return backend.configuration().simulator
+    backend_interface_version = _get_backend_interface_version(backend)
+    if backend_interface_version <= 1:
+        return backend.configuration().simulator
+    return False
```

Comment:
This function (`is_simulator_backend`) doesn't look especially accurate, but I'm guessing we're not really bothered about that?

Reply:
Well it's more that `BackendV2` doesn't have a `simulator` bool attribute in the same way as the `BackendConfiguration` does. It was never really an actionable flag anyway, if simulator is set to `True` what is a backend consumer to do with it?

That being said the code that uses this in the `QuantumInstance` just makes incorrect assumptions about it being aer xor basicar if this returns True and either can use a noise model, and also disable retry logic (which also is similarly not-portable).

Reply:
Haha, let's just leave it as-is then.